python/common/org/Python.java

```diff
@@ -814,15 +814,9 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
         } else if (class_or_type_or_tuple == null) {
             throw new org.python.exceptions.TypeError("isinstance expected 2 arguments, got 1");
         } else if (class_or_type_or_tuple instanceof org.python.types.Tuple) {
-            java.util.List<org.python.Object> classes = ((org.python.types.Tuple) class_or_type_or_tuple).value;
-            for (org.python.Object klass: classes) {
-                if (((org.python.types.Bool) org.Python.isinstance(object, klass).__bool__()).value) {
-                    return new org.python.types.Bool(true);
-                };
-            }
-            return new org.python.types.Bool(false);
+            return org.Python.issubclass(object.type(), class_or_type_or_tuple);
         } else if (class_or_type_or_tuple instanceof org.python.types.Type) {
-            return new org.python.types.Bool(object.type() == class_or_type_or_tuple);
+            return org.Python.issubclass(object.type(), class_or_type_or_tuple);
```

Comment:
Isn't it better to use `object.__class__` here instead of `object.type()` ?
That `Object.type()` public method should probably be deprecated.